File: C#/Projects/SnomedQuery.Parser/RF2Item.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SnomedQuery.Parser
{
  /// <summary>
  /// Base class for all RF2 items.
  /// </summary>
  public class RF2Item
  {
    /// <summary>
    /// Parser that created this node.
    /// </summary>
    public RF2Parser Parser;

    /// <summary>
    /// Snomed id.
    /// </summary>
    public Int64 Id;

    /// <summary>
    /// Parse string into boolean value.
    /// </summary>
    /// <param name="parser"></param>
    /// <param name="line"></param>
    /// <returns></returns>
    public static bool ParseBool(String s)
    {
      switch (s)
      {
        case "0": return false;
        case "1": return true;
        default: throw new NotImplementedException("Invalid boolean value");
      }
    }

    /// <summary>
    /// Parse effective time into c# date time instance.
    /// </summary>
    /// <param name="s"></param>
    /// <returns></returns>
    public static DateTime ParseEffectiveTime(String s)
    {
      if (s.Length != 8)
        throw new ApplicationException("Expected date of format YYYYMMDD");

      return new DateTime(Int32.Parse(s.Substring(0, 4)),
        Int32.Parse(s.Substring(4, 2)),
        Int32.Parse(s.Substring(6, 2)));
    }
  }
}

File: Java/output/SnomedQuery/Parser/RF2Item.java
package SnomedQuery.Parser;

  /** 
   Base class for all RF2 items.
  */
  public class RF2Item
  {
	/** 
	 Parser that created this node.
	*/
	public RF2Parser Parser;

	/** 
	 Snomed id.
	*/
	public long Id;

	/** 
	 Parse string into boolean value.
	 
	 @param parser
	 @param line
	 @return 
	*/
	public static boolean ParseBool(String s)
	{
	  switch (s)
	  {
		case "0":
			return false;
		case "1":
			return true;
		default:
			throw new UnsupportedOperationException("Invalid boolean value");
	  }
	}

	/** 
	 Parse effective time into c# date time instance.
	 
	 @param s
	 @return 
	*/
	public static java.time.LocalDateTime ParseEffectiveTime(String s)
	{
	  if (s.length() != 8)
	  {
		throw new RuntimeException("Expected date of format YYYYMMDD");
	  }

	  return java.time.LocalDateTime.of(Integer.parseInt(s.substring(0, 4)), Integer.parseInt(s.substring(4, 6)), Integer.parseInt(s.substring(6, 8)));
	}
  }